test/integration/api_test.lua

```diff
@@ -1,12 +1,264 @@
+local fio = require('fio')
+
 local t = require('luatest')
 local g = t.group('integration_api')
 
+local log = require('log')
+
 local helper = require('test.helper.integration')
-local cluster = helper.cluster
 
-g.test_sample = function()
-    local server = cluster.main_server
-    local response = server:http_request('post', '/admin/api', {json = {query = '{}'}})
-    t.assert_equals(response.json, {data = {}})
-    t.assert_equals(server.net_box:eval('return box.cfg.memtx_dir'), server.workdir)
+
+local http_client = require('http.client')
+local url = '/kv'
+
+
+g.before_all = function()
+    g.cluster = helper.shared.Cluster:new({
+        base_http_port = 8080,
+        base_advertise_port = 3000,
+        server_command = helper.shared.server_command,
+        datadir = helper.shared.datadir,
+        use_vshard = true,
+        replicasets = {
+            {
+                alias = 'router',
+                uuid = helper.shared.uuid('a'),
+                roles = {
+                    'api',
+                },
+                servers = {
+                    { instance_uuid = helper.shared.uuid('a', 1), alias = 'router' },
+                },
+            },
+            {
+                alias = 'kv-storage',
+                uuid = helper.shared.uuid('b'),
+                roles = {
+                    'kv-storage',
+                },
+                servers = {
+                    { instance_uuid = helper.shared.uuid('b', 1), alias = 'kv-storage-1-master' },
+                    { instance_uuid = helper.shared.uuid('b', 2), alias = 'kv-storage-1-replica-1' },
+                    { instance_uuid = helper.shared.uuid('b', 3), alias = 'kv-storage-1-replica-2' },
+                },
+            },
+        },
+        env = {
+            ['ENGINE'] = 'memtx',
+        },
+    })
+
+    g.cluster:start()
+    g.router = g.cluster.main_server
+end
+
+g.after_all = function()
+    g.cluster:stop()
+    fio.rmtree(g.cluster.datadir)
+end
+
+g.before_each(function()
+    for _, server in ipairs(g.cluster.servers) do
+        server.net_box:eval([[
+            local space = box.space.kv
+            if space ~= nil and not box.cfg.read_only then
+                space:truncate()
+            end
+        ]])
+    end
+end)
+
+g.after_each(function()
+
+end)
+
+-- CRUD helpers
+local function create(key, value)
+    return g.router:http_request('post', url..'/', {
+        json = {
+            key = key,
+            value = value,
+        },
+        http = { timeout = 1 },
+        raise = false,
+    })
+end
+
+local function read(key)
+    return g.router:http_request('get', url..'/'..key, { http = { timeout = 1 } , raise = false })
```

Comment:
The same as above.

BTW, aren't we have some "urljoin" functions anywhere in luatest?

Reply:
I searched but couldn't find std way to join url. So I wrote my own